bmwqemu.pm

```diff
@@ -99,7 +99,7 @@ sub save_vars {
     my $write_vars = \%vars;
     if ($args{no_secret}) {
         $write_vars = {};
-        $write_vars->{$_} = $vars{$_} for (grep !/^_SECRET_/, keys(%vars));
+        $write_vars->{$_} = $vars{$_} for (grep !/(^_SECRET_|_PASSWORD)/, keys(%vars));
```

Comment:
I'm thinking even better would be to make this the configurable default. Or make the backends add passwords to a different vars hash, if that's your goal, because that's the only way to be certain this works for all passwords used.

Reply:
> Consider what happens when cloning such jobs. Do they fail because the password is missing?
Also this. If backends use a different hash, a clone can fail with a clear error message like `variable FOO_PASSWORD not copied when cloning`

Reply:
I don't need this to be used for "all passwords" but just what we use within our backends.



>> Consider what happens when cloning such jobs. Do they fail because the password is missing?

> Also this. If backends use a different hash, a clone can fail with a clear error message like variable FOO_PASSWORD not copied when cloning

Not sure what you mean here. I don't want any clone to fail because we can't know what variables are defined within worker settings, which is commonly the case for such password variables to access remote machines.

Reply:
@kalikiana for security purposes I suggest to not make this a configurable variable. Otherwise it would be really hard to control who can change this variable which can also be abused for malicious intent.

Reply:
> Not sure what you mean here. I don't want any clone to fail because we can't know what variables are defined within worker settings, which is commonly the case for such password variables

We know what variables the backend requires e.g. `$bmwqemu::vars{IPMI_PASSWORD}` or `$bmwqemu::vars{NOVALINK_PASSWORD}`. The same ones that are listes in `doc/backend_vars.asciidoc`... actually, we seem to be missing HMC and ZVM... so what I'm saying is, if we know the variables we can make sure they are all covered or lead to a well-defined error.

Reply:
Discussed in the unblock call. I'll propose a PR for undocumented vars and we don't plan to add more variable names here